common/common-client/src/main/java/org/ow2/proactive/authentication/Connection.java

```diff
@@ -162,8 +165,16 @@ public T waitAndConnect(String url, long timeout) throws Exception {
             // waiting until scheduling is initialized
             while (leftTime > 0) {
                 long startTime = System.currentTimeMillis();
+                BooleanWrapper future = authentication.isActivated();
 
-                if (authentication.isActivated()) {
+                try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)